### ISSUE TITLE
Allow empty default values

### ISSUE THREAD

**Issue by vladimir-mencl-eresearch** [kreuzwerker/envplate#19]
Hi,

This is a follow-up to #15.  The fix for #15 allows specifying empty values in environment variables.

I think it would be neat if envPlate also allowed specifying empty default value, like:

    ${EMPTY_VARIABLE:-}

Right now, the parsing misbehaves, and envPlate reports:

    2016/01/21 03:20:32 [ ERROR ] 'test.txt' requires undeclared environment variable 'EMPTY_VARIABLE:-', no default is given

I.e., the :- becomes part of the variable name.

The part of the regexp parsing the default value has: ````(.+?)````.  I.e., it says "any character, at least once, all of that at least zero times".  Turns out, go regexp decides empty string does match this pattern and the ````:-```` becomes part of the variable name.

I can get around this by removing the ````+````, i.e.:

````
diff --git a/envplate.go b/envplate.go
index b2da37d..b81186d 100644
--- a/envplate.go
+++ b/envplate.go
@@ -15,7 +15,7 @@ const (
 	NotAnEscapeSequence = ""
 )

-var exp = regexp.MustCompile(`(\\*)\$\{(.+?)(?:\:\-(.+?))?\}`)
+var exp = regexp.MustCompile(`(\\*)\$\{(.+?)(?:\:\-(.?))?\}`)

 func Apply(globs []string) {

````

However, that still makes envPlate fail with:

    2016/01/21 03:35:07 [ ERROR ] 'test.txt' requires undeclared environment variable 'EMPTY_VARIABLE', no default is given

That is because the code interpreting the parsing results has no way to tell apart an empty default value specification from no default value specification.

This might be done by making the ````\:\-```` an explicit capture group and getting information whether the default value syntax was used.

Would you be happy to accept a pull-request based on this?

Cheers,
Vlad

**Comment by yawn** [comment]
Absolutely! This sounds very reasonable ...

**Comment by vladimir-mencl-eresearch** [comment]
Pull request sent (#20).

And, btw, I was off in my above post when elaborating the ````(.+?)```` part of the regexp - I now see the [Go regexp spec](https://golang.org/pkg/regexp/syntax/) defines ````x+?```` (and ````x*?````) as the _prefer-less_ versions of their counterparts without the question mark.  All makes sense now - and I got around the parsing problem by changing ````(.+?)```` to ````(.*?)````

**Comment by yawn** [comment]
Closed through #20
